### PR TITLE
FH-2745 - Fix Test Failures

### PR DESCRIPTION
**Pull request by TommyJ1994**
Adding Node 4.4 compatibility - fixing test failures.

### CODE REVIEW

**Comment by fh-automatron** [comment]
### Pull Request Quality Guidelines

As we continue to focus on quality, here are a few things to keep in mind:

* The code changes conform to the contributing & code style guides for this component (if they exist)
* A component owner should review your change (we've tagged these above)
* Automated testing has been added or updated for this change (e.g. UART, Unit/Acceptance tests)
* If updating server component this PR should be deployed to a grid somewhere (via `fhcap-cli`) - please provide a link to this
* Ensure JIRA is up to date before & after merge (e.g. deployment notes, tester notes, release notes, fix version)
* Any SonarQube issues should be addressed where applicable (use your own judgement)

Please ping a component owner (see `fh contrib`) when your builds are complete, changes are deployed and you're ready for a review.
Fhcap PR will be generated automatically.
To deploy your changes to a cluster place comment in this PR with pattern [x][y] where x=deploy and y=clustername you wish to deploy. 
You can also deploy your change manually using fhcap-cli

File: lib/cmd/common/build.js [hunk 1]
```diff
@@ -127,16 +127,18 @@ function validateArgsNgui(args) {
...
-    return payload + "&" + key + "=" + value;
-  }, initialPayload);
+  args.generateSrc=false;
+  return args;
+  // var initialPayload = "generateSrc=false";
+  // return _.reduce(args, function(payload, value, key) {
+  //   if (_.isUndefined(value)) {
+  //     return payload;
+  //   }
+  //   if (_.isObject(value)) {
```
Comment (fh-qube) [inline lib/cmd/common/build.js]: MAJOR Remove this commented out code. [rule](https://metrics.skunkhenry.com/coding_rules#rule_key=javascript%3ACommentedCode)

File: lib/cmd/common/build.js [hunk 2]
```diff
@@ -127,16 +127,18 @@ function validateArgsNgui(args) {
...
-    }
-    return payload + "&" + key + "=" + value;
-  }, initialPayload);
+  args.generateSrc=false;
+  return args;
+  // var initialPayload = "generateSrc=false";
+  // return _.reduce(args, function(payload, value, key) {
+  //   if (_.isUndefined(value)) {
+  //     return payload;
+  //   }
```
Comment (fh-qube) [inline lib/cmd/common/build.js]: MAJOR Remove this commented out code. [rule](https://metrics.skunkhenry.com/coding_rules#rule_key=javascript%3ACommentedCode)

File: lib/cmd/common/build.js [hunk 3]
```diff
@@ -127,16 +127,18 @@ function validateArgsNgui(args) {
...
+  //   if (_.isObject(value)) {
+  //     value = JSON.stringify(value);
+  //   }
```
Comment (fh-qube) [inline lib/cmd/common/build.js]: MAJOR Remove this commented out code. [rule](https://metrics.skunkhenry.com/coding_rules#rule_key=javascript%3ACommentedCode)

File: lib/cmd/common/build.js [hunk 4]
```diff
@@ -127,16 +127,18 @@ function validateArgsNgui(args) {
...
-  }, initialPayload);
+  args.generateSrc=false;
+  return args;
+  // var initialPayload = "generateSrc=false";
+  // return _.reduce(args, function(payload, value, key) {
+  //   if (_.isUndefined(value)) {
+  //     return payload;
+  //   }
+  //   if (_.isObject(value)) {
+  //     value = JSON.stringify(value);
```
Comment (fh-qube) [inline lib/cmd/common/build.js]: MAJOR Remove this commented out code. [rule](https://metrics.skunkhenry.com/coding_rules#rule_key=javascript%3ACommentedCode)

File: lib/cmd/common/build.js [hunk 5]
```diff
@@ -127,16 +127,18 @@ function validateArgsNgui(args) {
...
+  //   }
+  //   return payload + "&" + key + "=" + value;
+  // }, initialPayload);
```
Comment (fh-qube) [inline lib/cmd/common/build.js]: MAJOR Remove this commented out code. [rule](https://metrics.skunkhenry.com/coding_rules#rule_key=javascript%3ACommentedCode)

File: lib/cmd/common/build.js [hunk 6]
```diff
@@ -127,16 +127,18 @@ function validateArgsNgui(args) {
...
+  return args;
+  // var initialPayload = "generateSrc=false";
+  // return _.reduce(args, function(payload, value, key) {
+  //   if (_.isUndefined(value)) {
+  //     return payload;
+  //   }
+  //   if (_.isObject(value)) {
+  //     value = JSON.stringify(value);
+  //   }
+  //   return payload + "&" + key + "=" + value;
```
Comment (fh-qube) [inline lib/cmd/common/build.js]: MAJOR Remove this commented out code. [rule](https://metrics.skunkhenry.com/coding_rules#rule_key=javascript%3ACommentedCode)

File: lib/cmd/common/build.js [hunk 7]
```diff
@@ -127,16 +127,18 @@ function validateArgsNgui(args) {
...
-    }
-    if (_.isObject(value)) {
-      value = JSON.stringify(value);
-    }
-    return payload + "&" + key + "=" + value;
-  }, initialPayload);
+  args.generateSrc=false;
+  return args;
+  // var initialPayload = "generateSrc=false";
+  // return _.reduce(args, function(payload, value, key) {
```
Comment (fh-qube) [inline lib/cmd/common/build.js]: MAJOR Remove this commented out code. [rule](https://metrics.skunkhenry.com/coding_rules#rule_key=javascript%3ACommentedCode)

File: lib/cmd/common/build.js [hunk 8]
```diff
@@ -127,16 +127,18 @@ function validateArgsNgui(args) {
...
-      return payload;
-    }
-    if (_.isObject(value)) {
-      value = JSON.stringify(value);
-    }
-    return payload + "&" + key + "=" + value;
-  }, initialPayload);
+  args.generateSrc=false;
+  return args;
+  // var initialPayload = "generateSrc=false";
```
Comment (fh-qube) [inline lib/cmd/common/build.js]: MAJOR Remove this commented out code. [rule](https://metrics.skunkhenry.com/coding_rules#rule_key=javascript%3ACommentedCode)

File: lib/cmd/common/build.js [hunk 9]
```diff
@@ -127,16 +127,18 @@ function validateArgsNgui(args) {
...
+  // return _.reduce(args, function(payload, value, key) {
+  //   if (_.isUndefined(value)) {
+  //     return payload;
```
Comment (fh-qube) [inline lib/cmd/common/build.js]: MAJOR Remove this commented out code. [rule](https://metrics.skunkhenry.com/coding_rules#rule_key=javascript%3ACommentedCode)

File: lib/cmd/common/build.js [hunk 10]
```diff
@@ -127,16 +127,18 @@ function validateArgsNgui(args) {
...
-    if (_.isObject(value)) {
-      value = JSON.stringify(value);
-    }
-    return payload + "&" + key + "=" + value;
-  }, initialPayload);
+  args.generateSrc=false;
+  return args;
+  // var initialPayload = "generateSrc=false";
+  // return _.reduce(args, function(payload, value, key) {
+  //   if (_.isUndefined(value)) {
```
Comment (fh-qube) [inline lib/cmd/common/build.js]: MAJOR Remove this commented out code. [rule](https://metrics.skunkhenry.com/coding_rules#rule_key=javascript%3ACommentedCode)

**Comment by fh-qube** [comment]
SonarQube analysis reported 10 issues:
* MAJOR 10 major

Watch the comments in this conversation to review them.

**Comment by TommyJ1994** [comment]
ping @wei-lee

File: package.json [hunk 11]
```diff
@@ -61,13 +61,13 @@
...
     "load-grunt-tasks": "0.4.0",
-    "nock": "5.0.0",
+    "nock": "8.0.0",
     "proxyquire": "0.4.1",
     "time-grunt": "0.3.1",
     "turbo-test-runner": "0.3.3"
   },
   "engines": {
-    "node": ">=0.4"
+    "node": "4.4"
```
Comment (wei-lee) [inline package.json]: we should not restrict the version of node here. fhc is running on user machines and they could have different versions of node. Here we should just specify the minimum required node version. So we should keep the original requirement

**Comment by wei-lee** [comment]
One comment. LGTM

**Comment by wtrocki** [comment]
Retest this please! This build should pass.

**Comment by wtrocki** [comment]
@TommyJ1994 Investigated issue. There are multiple issues with unit tests that are now breaking build on node.10:
- plato
- jshint (should be replaced)
- zatana

We definietly can remove plato and jshint from dependencies, but I also feel that we can resolve problem by using different aproach.
Jenkins jobs were configured to work with server side modules where version is strictly defined. 
For fhc we should always build to work with multiple versions. 
I think that we should build it (test) with the latest version. To do that we would need to introduce different nvm setup that would be used for client side modules. See:

https://github.com/fheng/jenkins-bob-builder/blob/052a1f3e8caee3491fb77a0451430740b015e183/jobs/global-defaults.yaml#L162-L170

**Pull request merged**